spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -229,40 +215,23 @@ private void commitOperation(SnapshotUpdate<?> operation, String description) {
 
   private void abort(WriterCommitMessage[] messages) {
     if (cleanupOnAbort) {
-      Map<String, String> props = table.properties();
-      Tasks.foreach(files(messages))
-          .executeWith(ThreadPools.getWorkerPool())
-          .retry(PropertyUtil.propertyAsInt(props, COMMIT_NUM_RETRIES, COMMIT_NUM_RETRIES_DEFAULT))
-          .exponentialBackoff(
-              PropertyUtil.propertyAsInt(
-                  props, COMMIT_MIN_RETRY_WAIT_MS, COMMIT_MIN_RETRY_WAIT_MS_DEFAULT),
-              PropertyUtil.propertyAsInt(
-                  props, COMMIT_MAX_RETRY_WAIT_MS, COMMIT_MAX_RETRY_WAIT_MS_DEFAULT),
-              PropertyUtil.propertyAsInt(
-                  props, COMMIT_TOTAL_RETRY_TIME_MS, COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT),
-              2.0 /* exponential */)
-          .throwFailureWhenFinished()
-          .run(
-              file -> {
-                table.io().deleteFile(file.path().toString());
-              });
+      SparkCleanupUtil.deleteFiles("job abort", table.io(), files(messages));
     } else {
-      LOG.warn(
-          "Skipping cleaning up of data files because Iceberg was unable to determine the final commit state");
+      LOG.warn("Skipping cleanup of written files, unable to determine the final commit state");
     }
   }
 
-  private Iterable<DataFile> files(WriterCommitMessage[] messages) {
-    if (messages.length > 0) {
-      return Iterables.concat(
-          Iterables.transform(
-              Arrays.asList(messages),
-              message ->
-                  message != null
-                      ? ImmutableList.copyOf(((TaskCommit) message).files())
-                      : ImmutableList.of()));
-    }
-    return ImmutableList.of();
+  private List<DataFile> files(WriterCommitMessage[] messages) {
```

Comment:
I need a list to know the collection size.

Reply:
I have a little concern about memory, now we are manifesting paths into List, instead of keeping them as Iterable (if they are originally).  I see its mostly to log, I wonder if we can't implement a wrapping counter iterable for that?

Reply:
I'm okay either way, it seems like we were previously anyways materializing the WriterCommitMessages which have the files anyways? using s3 as an example, it takes 1 million objects with the worst case key length of 1024 bytes to use 1 GB of memory.

Reply:
That's true.  actually @amogh-jahagirdar was wondering if you know, is there a reason we dont have the deleteFiles() return number of files?  Would be probably be more convenient for callers to log the size that way?

Reply:
I changed the code to keep a list of files (shouldn't cost anything extra as those files are already there) and switched to using `Lists.transform()`, which is a lazy transform in `SparkCleanupUtil`.

Reply:
@szehon-ho @amogh-jahagirdar, could you take another look?